src/main/ngapp/accio/src/app/editor/magic-wand.service.ts

```diff
@@ -298,12 +212,54 @@ export class MagicWandService {
         }
         // Visits the pixel and check if it should be part of the mask.
         visited.add(this.coordToDataArrayIndex(x, y, imgData.width));
-        if (this.getIsScribbleMask(scribbles, imgData, neighborPixel, tolerance)) {
+        // if (this.getIsScribbleMask(scribbles, imgData, neighborPixel, tolerance)) {
+        //   visit.push(neighborPixel);
+        // }
+        if (tree.nearest(
+            this.scribblesToColors(new Set([
+            this.coordToDataArrayIndex(x, y, imgData.width)]), imgData)[0],
```

Comment:
It's pretty confusing to use the scribblesToColors function this way. I would suggest having a different function that turns a pixel index into a Color object that you can call here. scribblesToColors can also call that function.

Reply:
Thanks for pointing this out! I then realized that I already have an existing function that does this, I just need to modify it a bit.